packages/http/libraries/core/src/http/adapter/InversifyHttpAdapter.ts

```diff
@@ -540,7 +525,7 @@ export abstract class InversifyHttpAdapter<
       undefined;
     let httpStatusCode: HttpStatusCode | undefined = statusCode;
 
-    if (HttpResponse.is(value)) {
+    if (SuccessHttpResponse.is(value) || ErrorHttpResponse.is(value)) {
```

Comment:
```suggestion
    if (HttpResponse.is(value)) {
```

I see the benefit of adding `SuccessHttpResponse.is` and `ErrorHttpResponse.is`, but keeping `HttpResponse.is` would be perfect here, wouldn't it?

Reply:
what is preventing us from being a class? I mean, if we let HttpResponse as it was and we extend a `HttpResponse` class instead of implementing a `HttpResponse` interface, we got the best of both worlds, right?

Reply:
The problem is at `ErrorHttpResponse`. Now, `ErrorHttpResponse` is extending `Error` so It can not extend `HttpResponse`